### PR TITLE
Cran prep

### PR DESCRIPTION
**Pull request by JSchoenbachler**
Various test and formatting changes for cran release.

Also fixes the issue logged about saving the command versions of `prefetch` and `fasterq-dump`.

### CODE REVIEW

**Comment by JSchoenbachler** [comment]
@jakejh I'm getting some issues concerning the size of the tarball.

When I check on rhub, a note I get back is `Size of tarball: 15733812 bytes` which is a nudge saying it might be a large package.

Also, when I go to submit the package to be checked on Mac, I get an error and can't upload it-
```
Error in devtools::check_mac_release() : 
  Request Entity Too Large (HTTP 413). Failed to Uploading package.
```

Which I assume is also because the package tarball is large.

Looks like the largest items are test data items, and actually 3 files take up more than a third of the total size:

- GSE167458_eset.qs (1.2 MB)
- GSE50143_eset.qs (1.8 MB)
- E-MTAB-8714_eset.qs(4 MB)

Do you have any ideas on datasets we could use that might be more lightweight?

**Comment by JSchoenbachler** [comment]
Sticking to <= 5 MB is preferred, it seems.

**Comment by JSchoenbachler** [comment]
https://cran.r-project.org/web/packages/policies.html#
"Packages should be of the minimum necessary size. Reasonable compression should be used for data (not just .rda files) and PDF documentation: CRAN will if necessary pass the latter through qpdf.
As a general rule, neither data nor documentation should exceed 5MB (which covers several books). A CRAN package is not an appropriate way to distribute course notes, and authors will be asked to trim their documentation to a maximum of 5MB.

Where a large amount of data is required (even after compression), consideration should be given to a separate data-only package which can be updated only rarely (since older versions of packages are archived in perpetuity).

Similar considerations apply to other forms of “data”, e.g., .jar files."

**Comment by jakejh** [comment]
My latest commit should address the test data issue.

However, your code does not sufficiently address the issue of saving the info of prefetch and fasterq-dump, because the user could provide params that do not use the defaults.

**Comment by JSchoenbachler** [comment]
@jakejh sorry I haven't posted an update, I've been getting a weird thing when trying to check on rhub.

I see what you mean about the saving the info though, I'm working on that.

File: R/seeker.R [hunk 1]
```diff
@@ -506,6 +506,16 @@ seeker = function(params, parentDir = '.', dryRun = FALSE) {
 
   ####################
   fwrite(metadata, metadataPath)
+  commandVersions = checkDefaultCommands()
```
Comment (jakejh) [inline R/seeker.R]: This stuff should go in its own function. I also think commandVersions is not the most fitting name.

**Comment by JSchoenbachler** [comment]
See https://github.com/hugheylab/seeker/pull/32/commits/05fb23cb9a680083f33296186ffa17433104d8d5

**Comment by JSchoenbachler** [comment]
@jakejh for some reason, Windows rhub environment is erroring on check saying the `Biobase` version you specified isn't available. Can I bump it down or remove the version specified?

**Comment by jakejh** [comment]
Sure, you should be able to remove the version. It gets installed automatically by the other packages in Imports anyway.

**Comment by jakejh** [comment]
1. I'm going to harp on the naming. I still don't think "seeker commands" is fitting. Why not something along the lines of "sra toolkit commands"?
2. Please make the code consistent with lab style in all respects, so I don't find any trivial changes to make. Does the added function look like the other functions in the package?

**Comment by JSchoenbachler** [comment]
1. Currently it saves all commands, not just the toolkit ones. Do you want me to filter it down and update the name?
2. Will do, I'll check it.

**Comment by jakejh** [comment]
If it's currently saving all, why is it not checking for non-default values of the other commands?

**Comment by JSchoenbachler** [comment]
Good call, I'll update the function to handle those cases as well.

**Comment by jakejh** [comment]
@JSchoenbachler If the checks pass on my latest commits, please go through them carefully and notice the various ways I reduced the size of the test data.

Then hopefully you can edit cran_comments accordingly.

**Comment by JSchoenbachler** [comment]
@jakejh Doing a basic look through it looks like the size was reduced quite a bit, so I'll build it and see how large the tarball ends up being!

So far looks like the big ones are that you switched microarray tests and shaved off some of the mapping in the tx2gene included file.

**Comment by jakejh** [comment]
Yep. Do you also see how and why I revised the logging within the seeker function?

**Review by jakejh — APPROVED** [review]

**Comment by JSchoenbachler** [comment]
- "version" variable in the `getSysDeps` is renamed for clarity.
- Start logging index at 1 to match R standard.
- Grouped together the log index increment and function call to keep it better grouped.
- Revised to say step is "finished" rather than "complete".
- Moved step logging within the conditionals for steps so a step is only logged as being "finished" when it actually runs (my version would still log in the case of skipping).
- Adds log for "cleanup finished" to denote end of function call.

**Comment by JSchoenbachler** [comment]
Build size is now 4.2 MB, under the 5MB guideline! Running some checks on rhub and locally, then am I good to merge?

**Pull request merged**